docs: update testing document to include karma config

alan-agius4 opened the pull request:
In version 15, the Angular CLI will no longer generate a `karma.conf.js` by default.

This commit updates the testing document to include an example of how to include a custom `karma.conf.js`.  Also, we update several parts of the documentation which were outdated.

mary-poppins commented:
You can preview 0c0eb3c at https://pr47940-0c0eb3c.ngbuilds.io/.

mary-poppins commented:
You can preview f8296cb at https://pr47940-f8296cb.ngbuilds.io/.

mary-poppins commented:
You can preview 23338ae at https://pr47940-23338ae.ngbuilds.io/.

alan-agius4 committed:
docs: update testing document to include karma config

In version 15, the Angular CLI will no longer generate a `karma.conf.js` by default.

This commit updates the testing document to include an example of how to include a custom `karma.conf.js`.  Also, we update several parts of the documentation which were outdated.

dgp1130 commented on aio/content/guide/testing.md:
```diff
@@ -75,28 +73,89 @@ The tests run again, the browser refreshes, and the new test results appear.
-the `test.ts` files in the `src/` folder.
+    </code-example>
+
+1. In `angular.json`, configure the Karma builder to use the created configuration file.
```
This is an existing builder right? We should call out the new `karmaConfig` option in particular to draw attention to that as the change. Do we happen to have a `diff` type which gets rendered in AIO? That would be good to use here if it's supported.

alan-agius4 replied:
diff does not appear to be supported. I called out the karmaConfig option.

dgp1130 commented on aio/content/guide/testing.md:
```diff
@@ -75,28 +73,89 @@ The tests run again, the browser refreshes, and the new test results appear.
 
-The `karma.conf.js` file is a partial Karma configuration file.
-The CLI constructs the full runtime configuration in memory, based on application structure specified in the `angular.json` file, supplemented by `karma.conf.js`.
+  </code-example>
```
We do run the risk of these code samples falling out of date, but we shouldn't update them too often (majors/minors at most) so I'm not too worried about it. Something to keep in mind if we do end up changing these though.

alan-agius4 committed:
fixup! docs: update testing document to include karma config

alan-agius4 committed:
docs: remove CI testing setups

This section is unnecessary given that all is needed for the user is to run `ng test`, and is also tricky to maintain (Node was hard coded to v14 still).

We have no mechanism of testing or verifying these setups, and we have no motivation to do so.

Users should see their CI providers documentation Any preconfigured Node setup is going to suggest `ng test`, which should "just work" for Angular.

mary-poppins commented:
You can preview fc1102c at https://pr47940-fc1102c.ngbuilds.io/.

dgp1130 commented on aio/content/guide/testing.md:
```diff
@@ -197,202 +197,17 @@ next to their corresponding helper files.
 <code-example format="shell" language="shell">
 
-ng test --no-watch --no-progress --browsers=ChromeHeadlessCI
+ng test --no-watch --no-progress
```
Don't we do this automatically when the `CI` variable is set? Do we need to explicitly specify these options, or is this just for CI providers which don't set the `CI` environment variable?

dgp1130 replied:
Maybe this is something we should revisit then. It's always bugged me that `ng test` opened a visible browser and ran in watch mode. I was under the impression the `CI` variable at least make it work reasonably in this context. We can discuss this more and see if it's something we want to change.

jessicajaniuk reviewed (approved):
LGTM

AndrewKushnir commented:
This PR was merged into the repository by commit 665af467d88ad7bf5eb457617a2ce9f6963d22dd.
